test/test-agent-metadata.js

```diff
@@ -61,13 +57,40 @@ describe('agent interaction with metadata service', function() {
     }, 500);
   });
 
+  it('should preserve public interface when stopped', function(done) {
+      nock.disableNetConnect();
```

Comment:
The test above this one already checks the latter. However, it uses the private `stop()` method which does not affect the phantomness of the public agent, so manually calling stop doesn't work here.